apps/managedidentity/managedidentity_test.go

```diff
@@ -50,112 +61,326 @@ func getSuccessfulResponse(resource string) ([]byte, error) {
+		{name: "testAzureArcUserResourceAssigned", source: AzureArc, endpoint: imdsDefaultEndpoint, expectedSource: AzureArc, miType: UserAssignedResourceID("resourceId")},
+		{name: "testAzureArcUserObjectAssigned", source: AzureArc, endpoint: imdsDefaultEndpoint, expectedSource: AzureArc, miType: UserAssignedObjectID("objectId")},
+		{name: "testDefaultToImds", source: DefaultToIMDS, endpoint: imdsDefaultEndpoint, expectedSource: DefaultToIMDS, miType: SystemAssigned()},
+		{name: "testDefaultToImdsClientAssigned", source: DefaultToIMDS, endpoint: imdsDefaultEndpoint, expectedSource: DefaultToIMDS, miType: UserAssignedClientID("clientId")},
+		{name: "testDefaultToImdsResourceAssigned", source: DefaultToIMDS, endpoint: imdsDefaultEndpoint, expectedSource: DefaultToIMDS, miType: UserAssignedResourceID("resourceId")},
+		{name: "testDefaultToImdsObjectAssigned", source: DefaultToIMDS, endpoint: imdsDefaultEndpoint, expectedSource: DefaultToIMDS, miType: UserAssignedObjectID("objectId")},
+		{name: "testDefaultToImdsEmptyEndpoint", source: DefaultToIMDS, endpoint: "", expectedSource: DefaultToIMDS, miType: SystemAssigned()},
+		{name: "testDefaultToImdsLinux", source: DefaultToIMDS, endpoint: imdsDefaultEndpoint, expectedSource: DefaultToIMDS, miType: SystemAssigned()},
+		{name: "testDefaultToImdsEmptyEndpointLinux", source: DefaultToIMDS, endpoint: "", expectedSource: DefaultToIMDS, miType: SystemAssigned()},
 	}
 
 	for _, testCase := range testCases {
-		t.Run(http.StatusText(testCase.code), func(t *testing.T) {
-			fakeErrorClient := mock.Client{}
-			responseBody, err := makeResponseWithErrorData(testCase.err, testCase.desc)
+		t.Run(string(testCase.source), func(t *testing.T) {
+			unsetEnvVars(t)
+			setEnvVars(t, testCase.source)
+			setCustomAzureArcFilePath(t, fakeAzureArcFilePath)
+
+			actualSource, err := GetSource(testCase.miType)
 			if err != nil {
-				t.Fatalf("error while forming json response : %s", err.Error())
+				t.Fatalf("error while getting source: %s", err.Error())
 			}
-			fakeErrorClient.AppendResponse(mock.WithHTTPStatusCode(testCase.code),
-				mock.WithBody(responseBody))
-			client, err := New(SystemAssigned(), WithHTTPClient(&fakeErrorClient))
+
+			if actualSource != testCase.expectedSource {
+				t.Errorf(errorExpectedButGot, testCase.expectedSource, actualSource)
+			}
+		})
+	}
+}
+
+func TestAzureArcReturnsWhenHimdsFound(t *testing.T) {
+	if runtime.GOOS == "darwin" {
+		t.Skip("Skipping test on macOS as HIMDS is not supported")
+	}
+
+	testCases := []struct {
+		name           string
+		source         Source
+		endpoint       string
+		expectedSource Source
+		miType         ID
+	}{
+		{name: "testAzureArcSystemAssigned", source: AzureArc, endpoint: "imdsDefaultEndpoint", expectedSource: AzureArc, miType: SystemAssigned()},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(string(testCase.source), func(t *testing.T) {
+			unsetEnvVars(t)
```

Comment:
This test is passing for me locally and on linux / windows vms. What is going wrong for you?
The test case doesn't need env its just to check that we return true for getting himds when on endpoints are sent

Reply:
But it doesn't set IDENTITY_ENDPOINT or IMDS_ENDPOINT or fake the HIMDS path. I'm surprised it passes on a machine not managed by Arc. It fails like this on my machine:
```
=== FAIL: apps/managedidentity TestAzureArcReturnsWhenHimdsFound/AzureArc (0.00s)
    managedidentity_test.go:205: expected AzureArc, got DefaultToIMDS
```

Reply:
Sorry I meant to say its passing on linux and windows VMs, not locally since I'm on Mac.
I guess it not setting identityEndpoint and imdsEndpoint is ok since the first if is checking if they are not empty to return true, otherwise continue on. I'm guessing the VMs have himds in whatever location its checking and thats why, let me look more

Reply:
Can you try again, should work now